### PR TITLE
fix: Store blocks_validated in DB for Stability Validators

### LINKED ISSUE #12281: Stability validators timeout
## Description

`https://explorer.testnet.stabilityprotocol.com/api/v2/validators/stability` fails with 524

## Version

[v8.0.1-alpha.1](https://github.com/blockscout/blockscout/tree/v8.0.1-alpha.1)

### PR DESCRIPTION
**Pull request by nikitosing**
Closes #12281 

## Changelog
- Store `blocks_validated` in DB for Stability Validators

## Checklist for your Pull Request (PR)

- [ ] If I added new functionality, I added tests covering it.
- [ ] If I fixed a bug, I added a regression test to prevent the bug from silently reappearing again.
- [ ] I checked whether I should update the docs and did so by submitting a PR to [docs repository](https://github.com/blockscout/docs).
- [ ] If I added/changed/removed ENV var, I submitted a PR to [docs repository](https://github.com/blockscout/docs) to update the list of [env vars](https://github.com/blockscout/docs/blob/master/setup/env-variables/README.md) and I updated the version to `master` in the Version column. If I removed variable, I added it to [Deprecated ENV Variables](https://github.com/blockscout/docs/blob/master/setup/env-variables/deprecated-env-variables/README.md) page. After merging docs PR, changes will be reflected in these [pages](https://docs.blockscout.com/setup/env-variables).
- [ ] If I added new DB indices, I checked, that they are not redundant, with PGHero or other tools.
- [ ] If I added/removed chain type, I modified the Github CI matrix and PR labels accordingly.

## Summary by CodeRabbit

- **New Features**
  - Introduced a persistent counter for the number of blocks validated by each stability validator.
  - Stability validators' block counts are updated automatically during block import and reflected in the database.
  - Added parsing and importing support for stability validators in the block fetching process.
- **Bug Fixes**
  - Ensured accurate, atomic updates to validator counters, including handling multiple and zero increments.
- **Tests**
  - Added comprehensive tests covering validator counter updates during import and block processing, including edge cases.
- **Chores**
  - Database migration adds and initializes the new validator counter field.
  - Updated factory and test setups to support the new validator counter feature.
- **Refactor**
  - Simplified sorting logic for stability validators by using direct field sorting instead of dynamic queries.
  - Improved API test clarity by directly setting validator block counts instead of simulating via associated records.

### CODE REVIEW

File: apps/indexer/test/indexer/block/catchup/fetcher_test.exs [hunk 1]
```diff
@@ -694,6 +694,167 @@ defmodule Indexer.Block.Catchup.FetcherTest do
 
       assert %{from_number: 1, to_number: 0} = Repo.one(MissingBlockRange)
     end
+
+    if Application.compile_env(:explorer, :chain_type) == :stability do
+      test "update stability validator counter", %{
+        json_rpc_named_arguments: json_rpc_named_arguments
+      } do
+        Application.put_env(:indexer, Indexer.Block.Catchup.Fetcher, batch_size: 1, concurrency: 10)
+        Application.put_env(:indexer, :block_ranges, "1..2")
```
Comment (sl1depengwyn) [inline apps/indexer/test/indexer/block/catchup/fetcher_test.exs:703]: Please, restore env state in on_exit callback

File: apps/explorer/lib/explorer/chain/stability/validator.ex [hunk 2]
```diff
@@ -218,7 +212,7 @@ defmodule Explorer.Chain.Stability.Validator do
   @spec insert_validators([map()]) :: {non_neg_integer(), nil | []}
   def insert_validators(validators) do
     Repo.insert_all(__MODULE__, validators,
-      on_conflict: {:replace_all_except, [:inserted_at]},
+      on_conflict: {:replace_all_except, [:inserted_at, :blocks_validated]},
```
Comment (sl1depengwyn) [inline apps/explorer/lib/explorer/chain/stability/validator.ex:214]: Why we don't want to update blocks_validated field?

File: apps/indexer/test/indexer/block/catchup/fetcher_test.exs (hunk 1 above)
Comment (nikitosing) [reply apps/indexer/test/indexer/block/catchup/fetcher_test.exs:703]: It's already handled https://github.com/blockscout/blockscout/blob/743062d9f811d153c44bde2a9d7effd11f1d4343/apps/indexer/test/indexer/block/catchup/fetcher_test.exs#L150-L152

File: apps/explorer/lib/explorer/chain/stability/validator.ex (hunk 2 above)
Comment (nikitosing) [reply apps/explorer/lib/explorer/chain/stability/validator.ex:214]: Because this function is used in `Indexer.Fetcher.Stability.Validator` which is used for updating list of validators based on the RPC calls, in that fetcher updated only `state` field, `blocks_validated` counter updates on regular index

**Review by sl1depengwyn — APPROVED** [review]

File: apps/explorer/lib/explorer/chain/stability/validator.ex [hunk 3]
```diff
@@ -5,7 +5,8 @@ defmodule Explorer.Chain.Stability.Validator do
 
   use Explorer.Schema
 
-  alias Explorer.Chain.{Address, Import}
+  alias BlockScoutWeb.GraphQL.Resolvers.Block
+  alias Explorer.Chain.{Address, Block, Import}
```
Comment (vbaranov) [inline apps/explorer/lib/explorer/chain/stability/validator.ex]: Do we need the alias:
`alias BlockScoutWeb.GraphQL.Resolvers.Block`
in this module?

It seems we use `alias Explorer.Chain.Block` in it.

Comment (nikitosing) [reply apps/explorer/lib/explorer/chain/stability/validator.ex]: Yes, nice catch! Fixed in 873c991d06a491c234ca7b860a2d23315a9ac24e
